tsdb/wal/wal.go

```diff
@@ -913,6 +917,10 @@ func (r *segmentBufReader) Close() (err error) {
 
 // Read implements io.Reader.
 func (r *segmentBufReader) Read(b []byte) (n int, err error) {
+	if r.segs == nil {
```

Comment:
@codesome is this what you meant in https://github.com/prometheus/prometheus/issues/9605#issuecomment-962429920

I feel this check for every `Read` may be an overkill but couldn't think of a  simpler better way as well. 
May be we could return an emptySegmentReader struct that implements `io.ReaderCloser` so that 
`NewSegmentsRangeReader` becomes ...

```
func NewSegmentsRangeReader(sr ...SegmentRange) (io.ReadCloser, error) {   
  ...
  for ... {
    ...
    segs = append(segs, s)     
  }
  
  if len(segs) == 0 {
    return &emptySegmentReader{}
  }
  return NewSegmentBufReader(segs...), nil     
}
```

Reply:
This is not an overkill, and I suggest using the length
```suggestion
	if len(r.segs) == 0 {
```